### PR TITLE
PLT-1360/generator library for marconi testing 

### PR DESCRIPTION
**Pull request by kayvank**
- Refactor the generators used in marconi unit tests into a marconi library so that we can share them among marconi indexers.
- Moved all Marconi Indexer tests to marconi/test/Test/Marconi/Index folder

Pre-submit checklist:
- Branch
    - [ ] Tests are provided (if possible)
    - [ ] Commit sequence broadly makes sense
    - [ ] Key commits have useful messages
    - [ ] Formatting, PNG optimization, etc. are updated
    - [ ] Important changes are reflected in changelog.d of the affected packages
- PR
    - [ ] Self-reviewed the diff
    - [ ] Useful pull request description
    - [ ] Reference the ADR in the PR and reference the PR in the ADR (if revelant)
    - [ ] Reviewer requested

### CODE REVIEW

File: marconi/marconi.cabal [hunk 1]
```diff
@@ -131,6 +153,43 @@ executable marconi
     , directory
     , prettyprinter
 
+library marconi-test-lib
+  import:          lang
+  visibility:      public
+  hs-source-dirs:  test-lib
+  exposed-modules:
+    Gen.Marconi.Types
```
Comment (kayvank) [inline marconi/marconi.cabal:161]: Naming convention is similar to that of Cardno-Api, are we OK with that?

File: marconi/test-lib/Gen/Marconi/Types.hs [hunk 2]
```diff
@@ -0,0 +1,75 @@
+{-# LANGUAGE RecordWildCards #-}
+
+module Gen.Marconi.Types
+  ( genSlotNo
```
Comment (kayvank) [inline marconi/test-lib/Gen/Marconi/Types.hs:4]: These are the generators used in Utxo related tests. I'll move @koslambrou generators here as well and refactor the commonalities if we're OK with this approach

File: marconi/marconi.cabal (hunk 1 above)
Comment (koslambrou) [reply marconi/marconi.cabal:161]: :+1:

File: marconi/test-lib/Gen/Marconi/Types.hs [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+--   let (_inlineScript, _inlineScriptHash)=  Utxo.getRefScriptAndHash script
+--   pure $ Utxo.Utxo {..}
+
+-- genShelleyEvents :: Gen (Utxo.StorableEvent Utxo.UtxoHandle)
+-- genShelleyEvents = do
+--   ueUtxos <- Gen.set (Range.linear 1 3) genShelleyUtxo
+--   ueInputs <- Gen.set (Range.linear 1 2) CGen.genTxIn
+--   ueBlockNo <- genBlockNo
+--   ueChainPoint <- genChainPoint
+--   pure $ Utxo.UtxoEvent {..}
```
Comment (koslambrou) [inline marconi/test-lib/Gen/Marconi/Types.hs:120]: Bin?

**Review by koslambrou — APPROVED** [review]

File: marconi/test-lib/Gen/Marconi/Types.hs (hunk 2 above)
Comment (koslambrou) [reply marconi/test-lib/Gen/Marconi/Types.hs]: :+1:

**Pull request closed without merging**